src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ApplicationPropertyPage/ResourceSpecificationKindValueProvider.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Properties
+{
+    [ExportInterceptingPropertyValueProvider("ResourceSpecificationKind", ExportInterceptingPropertyValueProviderFile.ProjectFile)]
+    internal sealed class ResourceSpecificationKindValueProvider : InterceptingPropertyValueProviderBase
+    {
+        // TODO should the rule file generate property and enum value constants that we can use here instead of these string literals?
+
+        public override async Task<string?> OnSetPropertyValueAsync(string propertyName, string unevaluatedPropertyValue, IProjectProperties defaultProperties, IReadOnlyDictionary<string, string>? dimensionalConditions = null)
+        {
+            if (unevaluatedPropertyValue == "Win32Resource")
```

Comment:
Should this be "IconAndFile"?

Reply:
Yes it looks like I put this and the one below backwards. Can you fix this up, or give me write access to your fork (I don't often force push, but when I do I always use with-lease 😄).

Reply:
I'll fix it up.